lib/routes/routeBackbeat.js

```diff
@@ -304,12 +304,15 @@ function putMetadata(request, response, bucketInfo, objMd, log, callback) {
             // use original data locations
             omVal.location = objMd.location;
         }
-        // specify both 'versioning' and 'versionId' to create a "new"
-        // version (updating master as well) but with specified
-        // versionId
+        // Specify both 'versioning' and 'versionId' to create a "new" version
+        // (updating master as well) but with specified versionId. If the object
+        // is from a source bucket without versioning (i.e. NFS), then we want
+        // to create a version for the replica object even though none was
+        // provided in the object metadata value.
         const options = {
             versioning: omVal.versionId !== undefined ||
-                omVal.replicationInfo.isNFS === true,
+                (omVal.replicationInfo.isNFS === true &&
+                omVal.replicationInfo.status === 'REPLICA'),
```

Comment:
Previously, the versioning field was unconditionally set to true.
If the purpose of the changeset is to prevent the agent from making a put with versioning on the source object when it has the attribute replicationInfo.isNFS set to true, and only that particular case. I think the following is more accurate:

```
versioning: omVal.replicationInfo.isNFS !== true || omVal.replicationInfo.status === 'REPLICA'
```

Reply:
in addition to this. the omVal.replicationInfo.isNFS must be cleared if omVal.replicationInfo.status === 'REPLICA' is true

Reply:
> The versioning field was set to true on the condition that a version ID was provided in the omVal
You changed this in b8a2c8ede5ce610f10fb6d887c6e729ec4bd23c6
originally it was unconditionally set to true

the exception is: not put with versioning the source objectMD when isNFS is true on that object. which is what says the expression I put in my previous comment

Reply:
Ah, yes indeed this simplifies the condition. Updated!